framework/src/materials/MaterialData.C

```diff
@@ -70,6 +71,50 @@ MaterialData::reinit(const std::vector<std::shared_ptr<Material>> & mats)
     mat->computeProperties();
 }
 
+void
+MaterialData::reinitDotProps(Real dt, Real dt_old, Moose::TimeIntegratorType scheme)
+{
+  for (unsigned int i = 0; i < _dot_props.size(); ++i)
+  {
+    MaterialProperty<Real> & dot_prop = *_dot_props[i];
+    const MaterialProperty<Real> & old_prop = *_old_props[i];
+    const MaterialProperty<Real> & older_prop = *_older_props[i];
+    const MaterialProperty<Real> & prop = *_current_props[i];
+
+    switch (scheme)
```

Comment:
This isn't the right place for this.  If you add a new time integrator... the properties won't get integrated properly.

Reply:
The problem is that this evaluation has to go with material property evaluations. Time integrators do not provide a place for holding this calculation though.

Reply:
We can add a new function in `TimeIntegrator`, which does this evaluation and can be called in `FEProblemBase::reinitMaterials`. Let me change this PR a little.